packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -131,6 +132,14 @@ export class RouterliciousRestWrapper extends RestWrapper {
 		statusCode: number,
 		canRetry = true,
 	): Promise<IR11sResponse<T>> {
+		// Check whether this request has been made before or if it is a retry.
+		const originalRequestUrl = requestConfig.url;
+		const requestKey = `${requestConfig.method ?? ""}|${requestConfig.url ?? ""}`;
+		const requestRetryCount = this.retryCounter.get(requestKey);
+		if (requestRetryCount) {
+			requestConfig.url = `${requestConfig.url}&retryCount=${requestRetryCount}`;
```

Comment:
What if this url already has query params?

Reply:
Can we use the `URL` and `URLSearchParams` APIs here to make sure the formatting of the url is correct here?